hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -0,0 +1,380 @@
+            throws PreCheckException {
+
+        final var ownerAccount = accountStore.getAliasedAccountById(ownerId);
+        validateTruePreCheck(ownerAccount != null, INVALID_OWNER_ID);
+
+        // If the sender account is immutable, then we throw an exception.
+        final var key = ownerAccount.key();
+        if (key == null || !isValid(key)) {
+            throw new PreCheckException(ACCOUNT_IS_IMMUTABLE);
+        }
+        context.requireKey(key);
+    }
+
+    @SuppressWarnings("java:S2259")
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn, "Transaction body cannot be null");
+        final var op = txn.tokenRejectOrThrow();
+
+        validateFalsePreCheck(op.rejections().isEmpty(), INVALID_TRANSACTION_BODY);
+        if (op.hasOwner()) {
+            validateAccountID(op.owner(), null);
+        }
+
+        final var uniqueTokenReferences = new HashSet<TokenReference>();
+        for (final var rejection : op.rejections()) {
+            if (!uniqueTokenReferences.add(rejection)) {
+                throw new PreCheckException(TOKEN_REFERENCE_REPEATED);
+            }
+            // Ensure one token type per single rejection reference.
+            validateFalsePreCheck(rejection.hasFungibleToken() && rejection.hasNft(), INVALID_TRANSACTION_BODY);
+
+            if (rejection.hasFungibleToken()) {
+                final var tokenID = rejection.fungibleToken();
+                validateTruePreCheck(tokenID != null && !tokenID.equals(TokenID.DEFAULT), INVALID_TOKEN_ID);
+            }
+            if (rejection.hasNft()) {
+                final var nftID = rejection.nft();
+                validateTruePreCheck(nftID != null && nftID.tokenId() != null, INVALID_NFT_ID);
+                validateTruePreCheck(nftID.serialNumber() > 0, INVALID_TOKEN_NFT_SERIAL_NUMBER);
+            }
+        }
+    }
+
+    @Override
+    public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
+        final var op = context.body().tokenRejectOrThrow();
+        final var rejectingAccountID = op.ownerOrElse(context.payer());
+        final var rejections = op.rejections();
+
+        final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.tokenRejectEnabled(), NOT_SUPPORTED);
+        validateTrue(rejections.size() <= ledgerConfig.tokenRejectsMaxLen(), TOKEN_REFERENCE_LIST_SIZE_LIMIT_EXCEEDED);
+
+        final var accountStore = context.writableStore(WritableAccountStore.class);
+        final var rejectingAccount = getIfUsableForAliasedId(
+                rejectingAccountID, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
```

Comment:
@Nana-EC  Are we expecting to allow aliases in `TokenReject` transaction? If so, we should let mirror node know that aliases are allowed here.

Reply:
yes we should.
It's always been a long term vision that aliases be used anywhere accountIds could be within reasons.
If there is a reason to hold back on aliases we should note it, or if we need to state this support we should also capture it.
Otherwise yeah we should inform mirror explicitly of this possibility.

Reply:
The TokenReject functionality supports aliases, indeed.

Reply:
Thanks, the mirror node will then also support aliases.